### PR TITLE
Address all IDE0032. Use auto-implemented property.

### PR DESCRIPTION
**Pull request by IEvangelist**
Supersedes part of https://github.com/dotnet/orleans/pull/8471

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/8614)

### CODE REVIEW

**Comment by ReubenBond** [comment]
Are these changes all generated automatically by code fixes? I need to work out what needs closer attention while reviewing and what doesn't. Eg, adding `readonly` to a field often results in the wrong behavior, so that will need special attention.

File: src/Orleans.Serialization/Session/ReferencedObjectCollection.cs [hunk 1]
```diff
@@ -41,7 +41,6 @@ public ReferencePair(uint id, object @object)
 
         private Dictionary<uint, object> _referenceToObjectOverflow;
         private Dictionary<object, uint> _objectToReferenceOverflow;
-        private uint _currentReferenceId;
```
Comment (ReubenBond) [inline src/Orleans.Serialization/Session/ReferencedObjectCollection.cs:44]: We should revert the changes in this file.

File: src/Orleans.Serialization/Buffers/Adaptors/SpanBufferWriter.cs [hunk 2]
```diff
@@ -9,7 +9,6 @@ namespace Orleans.Serialization.Buffers.Adaptors
     public struct SpanBufferWriter : IBufferWriter<byte>
     {
         private readonly int _maxLength;
-        private int _bytesWritten;
```
Comment (ReubenBond) [inline src/Orleans.Serialization/Buffers/Adaptors/SpanBufferWriter.cs:12]: We should revert the changes in this file.

File: src/Orleans.Serialization/Buffers/Adaptors/BufferSliceReaderInput.cs [hunk 3]
```diff
@@ -14,7 +14,6 @@ public struct BufferSliceReaderInput
     private static readonly SequenceSegment FinalSegmentSentinel = new();
     private readonly BufferSlice _slice;
     private SequenceSegment _segment;
-    private int _position;
```
Comment (ReubenBond) [inline src/Orleans.Serialization/Buffers/Adaptors/BufferSliceReaderInput.cs:17]: We should revert the changes in this file.

File: src/Orleans.Core/Messaging/PrefixingBufferWriter.cs [hunk 4]
```diff
@@ -56,8 +56,6 @@ internal sealed class PrefixingBufferWriter : IBufferWriter<byte>, IDisposable
         /// </summary>
         private Sequence privateWriter;
 
-        private int _committedBytes;
```
Comment (ReubenBond) [inline src/Orleans.Core/Messaging/PrefixingBufferWriter.cs:59]: We should revert the changes in this file.

**Review by ReubenBond — COMMENTED** [review]
Some reversions needed, otherwise LGTM

File: src/Orleans.Core.Abstractions/IDs/SystemTargetGrainId.cs [hunk 5]
```diff
@@ -14,20 +14,19 @@ namespace Orleans.Runtime
     public readonly struct SystemTargetGrainId : IEquatable<SystemTargetGrainId>, IComparable<SystemTargetGrainId>, ISpanFormattable
     {
         private const char SegmentSeparator = '+';
-        private readonly GrainId _grainId;
```
Comment (pentp) [inline src/Orleans.Core.Abstractions/IDs/SystemTargetGrainId.cs:17]: This should be reverted - it's stored as a field specifically to get better inlining (when accessing it through a property, the subfields cannot be directly addressed by the compiler unless the entire data structure is "inlined").

File: src/Orleans.Streaming/Common/PooledCache/CachedMessageBlock.cs [hunk 6]
```diff
@@ -19,7 +19,6 @@ public class CachedMessageBlock : PooledResource<CachedMessageBlock>
         private readonly CachedMessage[] cachedMessages;
         private readonly int blockSize;
         private int writeIndex;
-        private int readIndex;
```
Comment (pentp) [inline src/Orleans.Streaming/Common/PooledCache/CachedMessageBlock.cs:22]: This should be reverted also for performance & clarity.

**Review by pentp — CHANGES_REQUESTED** [review]

File: src/AdoNet/Shared/Storage/RelationalStorage.cs [hunk 7]
```diff
@@ -259,7 +247,7 @@ private RelationalStorage(string invariantName, string connectionString)
             CommandBehavior commandBehavior,
             CancellationToken cancellationToken)
         {
-            using (var connection = DbConnectionFactory.CreateConnection(_invariantName, _connectionString))
+            using (var connection = DbConnectionFactory.CreateConnection(invariantName, ConnectionString))
```
Comment (IEvangelist) [inline src/AdoNet/Shared/Storage/RelationalStorage.cs:250]:
```suggestion
            using (var connection = DbConnectionFactory.CreateConnection(_invariantName, ConnectionString))
```

**Pull request closed without merging**